packages/amplify_datastore/android/src/main/kotlin/com/amazonaws/amplify/amplify_datastore/types/model/FlutterModelSchema.kt

```diff
@@ -18,32 +18,33 @@ package com.amazonaws.amplify.amplify_datastore.types.model
 import com.amplifyframework.core.model.ModelSchema
 
 data class FlutterModelSchema(val map: Map<String, Any>) {
-    val name: String = map["name"] as String
-    private val pluralName: String = map["pluralName"] as String
-    private val authRules: List<FlutterAuthRule>? =
+    val name : String = map["name"] as String
+    private val pluralName : String? = map["pluralName"] as String?
+    private val authRules : List<FlutterAuthRule>? =
             (map["authRules"] as List<Map<String, Any>>?)?.map { serializedAuthRule ->
                 FlutterAuthRule(serializedAuthRule)
             }
-    private val fields: Map<String, FlutterModelField> =
-            (map["fields"] as Map<String, Any>).mapValues { entry ->
+    private val fields : Map<String, FlutterModelField>? =
+            (map["fields"] as Map<String, Any>?)?.mapValues { entry ->
                 FlutterModelField(entry.value as Map<String, Any>)
             }
-    private val associations: Map<String, FlutterModelAssociation>? =
-            (map["associations"] as Map<String, Any>?)?.mapValues { entry ->
-                FlutterModelAssociation(entry.value as Map<String, Any>)
-            }
-    private val indexes: Map<String, FlutterModelIndex>? =
-            (map["indexes"] as Map<String, Any>?)?.mapValues { entry ->
-                FlutterModelIndex(entry.value as Map<String, Any>)
+    private val associations : Map<String, FlutterModelAssociation>? =
+            ( fields )?.filterKeys { key -> fields[key]?.getModelAssociation() != null }
+                    ?.mapValues { entry ->
+                entry.value.getModelAssociation()!!
             }
 
     fun convertToNativeModelSchema(): ModelSchema {
         var builder: ModelSchema.Builder = ModelSchema.builder()
                 .name(name)
                 .pluralName(pluralName)
-                .fields(fields.mapValues { entry ->
-                    entry.value.convertToNativeModelField()
-                })
+
+        if(!fields.isNullOrEmpty()){
```

Comment:
Can fields be ever empty though? By design we require an Id field always.

Reply:
Right, I've made it non nullable now.  I had not considered the id field.